cot/src/response/into_response.rs

```diff
@@ -0,0 +1,258 @@
+impl_into_response_for_type_and_mime!(Bytes, mime::APPLICATION_OCTET_STREAM);
+
+impl<const N: usize> IntoResponse for &'static [u8; N] {
+    fn into_response(self) -> Response {
+        self.as_slice().into_response()
+    }
+}
+
+impl<const N: usize> IntoResponse for [u8; N] {
+    fn into_response(self) -> Response {
+        self.to_vec().into_response()
+    }
+}
+
+impl IntoResponse for Box<[u8]> {
+    fn into_response(self) -> Response {
+        Vec::from(self).into_response()
+    }
+}
+
+impl IntoResponse for BytesMut {
+    fn into_response(self) -> Response {
+        self.freeze().into_response()
+    }
+}
+
+// HTTP structures for common uses
+
+impl IntoResponse for StatusCode {
+    fn into_response(self) -> Response {
+        let mut res = ().into_response();
+        *res.status_mut() = self;
+        res
+    }
+}
+
+impl IntoResponse for http::HeaderMap {
+    fn into_response(self) -> Response {
+        let mut res = ().into_response();
+        *res.headers_mut() = self;
+        res
+    }
+}
+
+impl IntoResponse for http::Extensions {
+    fn into_response(self) -> Response {
+        let mut res = ().into_response();
+        *res.extensions_mut() = self;
+        res
+    }
+}
+
+impl IntoResponse for http::response::Parts {
+    fn into_response(self) -> Response {
+        Response::from_parts(self, Body::empty())
+    }
+}
+
+// Data type structures implementations
+
+impl IntoResponse for Html {
+    /// Create a new HTML response.
+    ///
+    /// This creates a new [`Response`] object with a content type of
+    /// `text/html; charset=utf-8` and given status code and body.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::html::Html;
+    /// use cot::response::IntoResponse;
+    ///
+    /// let html = Html::new("<div>Hello</div>");
+    ///
+    /// let response = html.into_response();
+    /// ```
+    fn into_response(self) -> Response {
+        self.as_str().to_owned().into_response().with_content_type(
+            http::header::HeaderValue::from_static(mime::TEXT_HTML_UTF_8.as_ref()),
+        )
+    }
+}
+
+// Shortcuts for common uses
+
+impl IntoResponse for Body {
+    fn into_response(self) -> Response {
+        Response::new(self)
+    }
+}
+
+impl<R> IntoResponse for (StatusCode, R)
+where
+    R: IntoResponse,
+{
+    fn into_response(self) -> Response {
+        let mut res = self.1.into_response();
+        *res.status_mut() = self.0;
+        res
+    }
+}
+
+impl<R> IntoResponse for (StatusCode, http::HeaderMap, R)
+where
+    R: IntoResponse,
+{
+    fn into_response(self) -> Response {
+        let mut res = self.2.into_response();
+        *res.status_mut() = self.0;
+        *res.headers_mut() = self.1;
+        res
+    }
+}
+
+impl<R> IntoResponse for (http::HeaderMap, R)
+where
+    R: IntoResponse,
+{
+    fn into_response(self) -> Response {
+        let mut res = self.1.into_response();
+        *res.headers_mut() = self.0;
+        res
+    }
+}
```

Comment:
Why do we need these if we have `with_header`, `with_status` and other methods?

Reply:
You can call it a leftover from the experimentation phase. I'm unsure which way is better: the trait methods or the above tuples. What do you think is better and should be retained?

Reply:
The trait methods are clearly better in my opinion:
* They are easily discoverable in the IDE (they will be listed among suggestions in the autocomplete popup)
* They are explicit, it's obvious what do they do because they have names
* There's no way you could try to use them in reverse order, as is the case with the tuples

I don't see a reason to keep the tuples because of this.